sktime/markovprocess/bhmm/hidden/src/bhmm_hidden_module.cpp

```diff
@@ -263,31 +244,28 @@ np_array<std::int32_t> samplePath(const np_array<dtype> &alpha, const np_array<d
     auto ABuf = A.data();
     {
         py::gil_scoped_release gil;
-        // initialize random number generator
-        if (seed == -1) {
-            srand(time(NULL));
-        } else {
-            srand(seed);
-        }
+        std::default_random_engine generator (seed);
 
         // Sample final state.
         for (std::size_t i = 0; i < N; i++) {
             psel[i] = alphaBuf[(T - 1) * N + i];
         }
 
+        std::discrete_distribution<> ddist (psel, psel+N);
         normalize(psel, psel + N);
         // Draw from this distribution.
-        path[T - 1] = _random_choice(psel, N);
+        path[T - 1] = ddist(generator); //_random_choice(psel, N);
 
         // Work backwards from T-2 to 0.
         for (std::size_t t = T - 1; t >= 1; t--) {
             // Compute P(s_t = i | s_{t+1}..s_T).
             for (std::size_t i = 0; i < N; i++) {
                 psel[i] = alphaBuf[(t - 1) * N + i] * ABuf[i * N + path[t]];
             }
-            normalize(psel, psel + N);
+            std::discrete_distribution<> ddistBackw (psel, psel+N);
```

Comment:
is it mandatory to recreate this object in the loop or is there a way to re-initialize it?

Reply:
there is a way to re-initialize it!

Reply:
muy bien!